From 6ec172c38be0f556a633848123edd30ece160ddc Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Thu, 5 Mar 2026 19:08:45 +0800
Subject: [PATCH 06/12] configurations: nvidia: add CX9 IO Mezz board
 configuration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add the configuration file for the NVIDIA CX9 IO Mezz board
used by VR NVL72 systems.

Tested:
```
root@bmc:~# busctl tree xyz.openbmc_project.EntityManager | grep "NVIDIA_VR_NVL72_IO_Board"
          ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_0
          │ └─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_0/NVIDIA_VR_NVL72_IO_Board_0_FRU
          ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_1
          │ └─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_1/NVIDIA_VR_NVL72_IO_Board_1_FRU
          ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_2
          │ └─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_2/NVIDIA_VR_NVL72_IO_Board_2_FRU
          ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_3
          │ └─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_IO_Board_3/NVIDIA_VR_NVL72_IO_Board_3_FRU
```

[1] Gerrit: https://gerrit.openbmc.org/c/openbmc/entity-manager/+/88082

Change-Id: Ib50a7d311683d48238b268b1b7a76c6eb6f67a22
Signed-off-by: Potin Lai <potin.lai@quantatw.com>
---
 configurations/meson.build                    |  1 +
 .../nvidia/cx9_mezzanine_module.json          | 26 +++++++++++++++++++
 2 files changed, 27 insertions(+)
 create mode 100644 configurations/nvidia/cx9_mezzanine_module.json

diff --git a/configurations/meson.build b/configurations/meson.build
index d6d509b..ec97351 100644
--- a/configurations/meson.build
+++ b/configurations/meson.build
@@ -297,6 +297,7 @@ configs = [
     'nvidia/clover_cx9.json',
     'nvidia/cx7_mezzanine_module.json',
     'nvidia/cx8_mezzanine_module.json',
+    'nvidia/cx9_mezzanine_module.json',
     'nvidia/gb200.json',
     'nvidia/gb300.json',
     'nvidia/imgx_connectx8_supernic_switch.json',
diff --git a/configurations/nvidia/cx9_mezzanine_module.json b/configurations/nvidia/cx9_mezzanine_module.json
new file mode 100644
index 0000000..7cdc62b
--- /dev/null
+++ b/configurations/nvidia/cx9_mezzanine_module.json
@@ -0,0 +1,26 @@
+{
+    "Exposes": [
+        {
+            "Address": "$address",
+            "Bus": "$bus",
+            "Name": "NVIDIA VR NVL72 IO Board $bus - 73 FRU",
+            "Type": "EEPROM"
+        }
+    ],
+    "Name": "NVIDIA VR NVL72 IO Board $bus - 73",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'ConnectX-9 800GE 2P NIC', 'BOARD_MANUFACTURER': '(Nvidia|NVIDIA)'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "BuildDate": "$BOARD_MANUFACTURE_DATE",
+        "Manufacturer": "$BOARD_MANUFACTURER",
+        "Model": "$BOARD_PRODUCT_NAME",
+        "PartNumber": "$BOARD_PART_NUMBER",
+        "SerialNumber": "$BOARD_SERIAL_NUMBER"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "$PRODUCT_ASSET_TAG"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Revision": {
+        "Version": "$PRODUCT_VERSION"
+    }
+}
-- 
2.52.0

